src/H5FDs3comms.c

```diff
@@ -1840,6 +1842,78 @@ H5FD__s3comms_load_aws_creds_from_file(FILE *file, const char *profile_name, cha
     FUNC_LEAVE_NOAPI(ret_value)
 } /* end H5FD__s3comms_load_aws_creds_from_file() */
 
+/*-----------------------------------------------------------------------------
+ *
+ * Function: H5FD__s3comms_load_aws_creds_from_env()
+ *
+ * Purpose:
+ *
+ *     Get aws credentials from environment variables AWS_ACCESS_KEY_ID,
+ *     AWS_SECRET_ACCESS_KEY, AWS_REGION and AWS_SESSION_TOKEN.
+ *     Values from these environment variables will override any values
+ *     for corresponding variables loaded from credentials and configuration
+ *     files.
+ *
+ *     Values for AWS_PROFILE and AWS_MAX_ATTEMPTS are not currently obtained.
+ *
+ * Return: SUCCEED/FAIL
+ *
+ */
+static herr_t
+H5FD__s3comms_load_aws_creds_from_env(char *key_id, char *secret_access_key, char *session_token,
+                                      char *aws_region)
+{
+    herr_t ret_value             = SUCCEED;
+    char  *key_id_env            = NULL;
+    char  *secret_access_key_env = NULL;
+    char  *session_token_env     = NULL;
+    char  *aws_region_env        = NULL;
+
+    FUNC_ENTER_PACKAGE
+
+    /* AWS_ACCESS_KEY_ID values are typically 16 or 20 characters, with up to 128 allowed.
+     * Difference in size between the one from the environment and one in cred files
+     * requires some special handling.
+     */
+    key_id_env = getenv("AWS_ACCESS_KEY_ID");
+    if (key_id_env != NULL && key_id_env[0] != '\0') {
+        if (strlen(key_id) == 0 || strncmp(key_id, key_id_env, strlen(key_id)) != 0)
+            strncpy(key_id, key_id_env, strlen(key_id_env));
+        key_id[strlen(key_id_env)] = '\0';
```

Comment:
Is there a reason to add a null-terminator to the existing `key_id` when it's not overwritten by the environment variable?